sdk_container/src/third_party/coreos-overlay/sys-apps/ignition/files/0020-usr-share-oem-oem.patch

```diff
@@ -0,0 +1,53 @@
+Date: Tue, 4 Apr 2023 12:12:42 +0200
+Subject: [PATCH 20/20] /usr/share/oem -> /oem
+
+---
+ config/util/translate.go    | 2 +-
+ docs/supported-platforms.md | 2 +-
+ internal/distro/distro.go   | 2 +-
+ 3 files changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/config/util/translate.go b/config/util/translate.go
+index 347d148c..d4c057b2 100644
+--- a/config/util/translate.go
++++ b/config/util/translate.go
+@@ -154,7 +154,7 @@ func FSGeneration(name string, fsMap map[string]string) (string, error) {
+ 	if _, ok := fsMap[name]; !ok {
+ 		addedSuffixCounter += 1
+ 		if name == "oem" {
+-			fsMap[name] = "/usr/share/oem"
++			fsMap[name] = "/oem"
+ 		} else {
+ 			// generate a new path
+ 			fsMap[name] = "/tmp/" + name + "-ign" + strconv.FormatUint(addedSuffixCounter, 10)
+diff --git a/docs/supported-platforms.md b/docs/supported-platforms.md
+index 90df6c6a..7867ba09 100644
+--- a/docs/supported-platforms.md
++++ b/docs/supported-platforms.md
+@@ -10,7 +10,7 @@ Ignition is currently only supported for the following platforms:
+ * [Amazon Web Services] (`aws`) - Ignition will read its configuration from the instance userdata. Cloud SSH keys are handled separately.
+ * [Microsoft Azure] (`azure`)- Ignition will read its configuration from the custom data provided to the instance. Cloud SSH keys are handled separately.
+ * [Microsoft Azure Stack] (`azurestack`) - Ignition will read its configuration from the custom data provided to the instance. Cloud SSH keys are handled separately.
+-* Bare Metal - Use the `ignition.config.url` kernel parameter to provide a URL to the configuration. The URL can use the `http://`, `https://`, `tftp://`, `s3://`, or `gs://` schemes to specify a remote config or the `oem://` scheme to specify a local config, rooted in `/usr/share/oem`.
++* Bare Metal - Use the `ignition.config.url` kernel parameter to provide a URL to the configuration. The URL can use the `http://`, `https://`, `tftp://`, `s3://`, or `gs://` schemes to specify a remote config or the `oem://` scheme to specify a local config, rooted in `/oem`.
+ * [Brightbox] (`brightbox`) - Ignition will read its configuration from the instance userdata. Cloud SSH keys are handled separately.
+ * `cloudsigma` - Ignition will read its configuration from the instance userdata. Cloud SSH keys are handled separately.
+ * [CloudStack] (`cloudstack`) - Ignition will read its configuration from the instance userdata via either metadata service or config drive. Cloud SSH keys are handled separately.
+diff --git a/internal/distro/distro.go b/internal/distro/distro.go
+index 644c42c4..84e575c7 100644
+--- a/internal/distro/distro.go
++++ b/internal/distro/distro.go
+@@ -33,7 +33,7 @@ var (
+ 	// initramfs directory containing distro-provided base config
+ 	systemConfigDir = "/usr/lib/ignition"
+ 	// initramfs directory to check before retrieving file from OEM partition
+-	oemLookasideDir = "/usr/share/oem"
++	oemLookasideDir = "/oem"
```

Comment:
I think this is the initrd directory that users can provide, therefore, it should stay at the same path.

Reply:
How can a user provide a custom directory into initrd? If I understand correctly, we provide initrd inside our kernel binary. Does this mean that we support custom built kernels with a custom initrd?

Reply:
One can provide multiple initrds in the boot loader and they get merged. This is actually what happens in the PXE boot with Flatcar where `flatcar_production_pxe_image.cpio.gz` is an additional initrd that brings the sqashfs image.

Reply:
Thanks, I'll revert this change. I must admit that this may be quite confusing having `/usr/share/oem` in initrd, but `/oem` in rootfs. I think this could be addressed with a documentation, but I dunno if we even have docs about this feature.

Reply:
Reverted.